modules/contenta_enhancements/contenta_enhancements.install

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/**
+ * @file
+ * Module install file.
+ */
+
+/**
+ * Implements hook_install().
+ */
+function contenta_enhancements_install() {
+  /** @var \Drupal\simple_oauth\Service\KeyGeneratorService $key_gen */
+  $key_gen = \Drupal::service('simple_oauth.key.generator');
+  /** @var \Drupal\simple_oauth\Service\Filesystem\Filesystem $file_system */
+  $file_system = \Drupal::service('simple_oauth.filesystem');
+  /** @var \Drupal\Core\Logger\LoggerChannelInterface $logger */
+  $logger = \Drupal::service('logger.channel.contentacms');
+
+  $keys_path = $file_system->realpath(DRUPAL_ROOT . '/../keys');
+  $pub_filename = sprintf('%s/public.key', $keys_path);
+  $pri_filename = sprintf('%s/private.key', $keys_path);
+
+  if ($file_system->fileExist($pub_filename) && $file_system->fileExist($pri_filename)) {
+    // 1. If the file already exists, then just set the correct permissions.
+    $file_system->chmod($pub_filename, 0600);
+    $file_system->chmod($pri_filename, 0600);
+    $logger->info('Key pair for OAuth 2 token signing already exists.');
+  }
+  else {
+    // 2. Generate the pair in the selected directory.
+    try {
+      $key_gen->generateKeys($keys_path);
+    }
+    catch (\Exception $e) {
+      // Unable to generate files after all.
+      $logger->error($e->getMessage());
+    }
+  }
+  // At this point the keys should be set. Now save the configuration.
```

Comment:
... In case there was an exception, you should not continue here, right?

Reply:
Good point